client/client.go

```diff
@@ -49,6 +50,15 @@ type Client interface {
 	Request(operation []byte) (resultChan <-chan []byte)
 }
 
+// ReplyCollector processes replies and extracts results
+//
+// Collect gathers Reply messages received from
+// the passed channel, finishes the request processing, and sends the
+// result of request execution to the passed channel.
+type ReplyCollector interface {
+	Collect(in <-chan *messages.Reply, out chan<- []byte, remover requestRemover)
```

Comment:
Note that this interface depends on MinBFT-specific definition of Reply message, as well as on unexported `requestRemover` type.

Reply:
I see two options for Reply message: either change the method signature to  `Collect(in <-chan interface{}, out chan<- []byte, remover requestRemover)` (which would require additional changes throughout the code) or simply adding an extra field to the Reply message (e.g., `metadata []byte`) that would allow this type to be more generic.

Regarding the unexported `requestRemover` type, I thought about changing into to be part of the external API, but since it is created using an object from the `client/internal` package, I thought it would be best to leave it unexported.

Reply:
> change the method signature to `Collect(in <-chan interface{}, out chan<- []byte, remover requestRemover)`

This seems to be more aligned with the whole idea of making implementation of internal connectivity generic.

> Regarding the unexported requestRemover type ... it would be best to leave it unexported.

I'm afraid external packages wouldn't be able to use the new interface because they have no access to the unexported type.

Reply:
> I'm afraid external packages wouldn't be able to use the new interface because they have no access to the unexported type.

Do you think it would be wise to make the RequestRemover external?